contracts/StateChannel.sol

```diff
@@ -444,11 +441,10 @@ contract StateChannel is Initializable, OwnableUpgradeable {
 
         // check is finish
         bool isFinish1 = query.isFinal;
-        bool isFinish2 = isFinish1 || amount == 0;
```

Comment:
[question] why remove the condition `amount == 0` to settle finalise the channel

Reply:
This is for automatically finalize if there is no amount when terminating, but it will has security issues and make it possible for the consumer to close the channel early (re-submit the old state) without paying.

Reply:
Make sense 👍